src/admissioncontrollerpolicy/admissionControllerPolicy.go

```diff
@@ -0,0 +1,294 @@
+			"operator": "Equals",
+			"value":    value,
+		}
+		existingSlice := preconditionMap["all"].([]interface{})
+		preconditionMap["all"] = append(existingSlice, newPrecondition)
+	}
+	return preconditionMap
+}
+
+func AutoGenPattern(templateKey string, pattern apiextensions.JSON) apiextensions.JSON {
+	newPattern := map[string]interface{}{
+		"spec": map[string]interface{}{
+			templateKey: pattern,
+		},
+	}
+	return newPattern
+}
+
+func ShouldSATokenBeAutoMounted(namespace string, labels types.LabelMap) bool {
+	client := cluster.ConnectK8sClient()
+
+	podList, err := client.CoreV1().Pods(namespace).List(context.TODO(), metav1.ListOptions{
+		LabelSelector: libs.LabelMapToString(labels),
+	})
+
+	if err != nil {
+		log.Warn().Msg(err.Error())
+		return true
+	}
+
+	if len(podList.Items) > 0 {
+		// Only inspect the first pod in the list as deployment pods have same behavior
+		pod := podList.Items[0]
+		containersSATokenMountPath, err := getSATokenMountPath(&pod)
+		if err != nil && strings.Contains(err.Error(), "service account token not mounted") {
+			log.Warn().Msg(err.Error())
+			return false
+		}
+		var sumResponses []*opb.Response
+		for _, container := range pod.Spec.Containers {
+			sumResp, err := obs.GetSummaryData(&opb.Request{
+				PodName:       pod.Name,
+				NameSpace:     pod.Namespace,
+				ContainerName: container.Name,
+				Type:          "process,file",
+			})
+			if err != nil {
+				log.Warn().Msg(err.Error())
+				return true
+			}
+			sumResponses = append(sumResponses, sumResp)
+		}
+		return serviceAccountTokenUsed(containersSATokenMountPath, sumResponses)
+	} else {
+		log.Warn().Msg("No pods found for the given labels")
+	}
+
+	return true
+}
+
+func getSATokenMountPath(pod *corev1.Pod) ([]containerMountPathServiceAccountToken, error) {
+	volumes := pod.Spec.Volumes
+	var tokenPath string
+	var projectedVolumeName string
+	var result = make([]containerMountPathServiceAccountToken, 0)
+	for _, volume := range volumes {
+		if volume.Projected != nil {
+			for _, projectedSources := range volume.Projected.Sources {
+				serviceAccountToken := projectedSources.ServiceAccountToken
+				if serviceAccountToken != nil {
+					tokenPath = serviceAccountToken.Path
+					projectedVolumeName = volume.Name
+					break
+				}
+			}
+		}
+	}
+
+	if tokenPath == "" || projectedVolumeName == "" {
+		return result,
+			fmt.Errorf("service account token not mounted for %s in namespace %s", pod.Name, pod.Namespace)
+	}
+
+	containers := pod.Spec.Containers
+	for _, container := range containers {
+		volumeMounts := container.VolumeMounts
+		for _, volumeMount := range volumeMounts {
+			if volumeMount.Name == projectedVolumeName {
+				result = append(result, containerMountPathServiceAccountToken{
+					podName:          pod.Name,
+					podNamespace:     pod.Namespace,
+					containerName:    container.Name,
+					saTokenMountPath: volumeMount.MountPath + string(filepath.Separator) + tokenPath,
+				})
+			}
```

Comment:
Assuming that we return only a single element in this function.
```suggestion
			if volumeMount.Name == projectedVolumeName {
				return containerMountPathServiceAccountToken{
					podName:          pod.Name,
					podNamespace:     pod.Namespace,
					containerName:    container.Name,
					saTokenMountPath: volumeMount.MountPath + string(filepath.Separator) + tokenPath,
				}
			}
```